be/src/runtime/stream_load/stream_load_pipe.cpp

```diff
@@ -23,7 +23,255 @@
 
 namespace starrocks {
 
-StreamLoadPipeInputStream::StreamLoadPipeInputStream(std::shared_ptr<StreamLoadPipe> file) : _pipe(std::move(file)) {}
+Status StreamLoadPipe::append(ByteBufferPtr&& buf) {
+    if (buf != nullptr && buf->has_remaining()) {
+        std::unique_lock<std::mutex> l(_lock);
+        // if _buf_queue is empty, we append this buf without size check
+        _put_cond.wait(l, [&]() {
+            return _cancelled || _buf_queue.empty() || _buffered_bytes + buf->remaining() <= _max_buffered_bytes;
+        });
+
+        if (_cancelled) {
+            return _err_st;
+        }
+        _buffered_bytes += buf->remaining();
+        _buf_queue.emplace_back(std::move(buf));
+        _get_cond.notify_one();
+    }
+    return Status::OK();
+}
+
+Status StreamLoadPipe::append(const char* data, size_t size) {
+    size_t pos = 0;
+    if (_write_buf != nullptr) {
+        if (size < _write_buf->remaining()) {
+            _write_buf->put_bytes(data, size);
+            return Status::OK();
+        } else {
+            pos = _write_buf->remaining();
+            _write_buf->put_bytes(data, pos);
+
+            _write_buf->flip();
+            RETURN_IF_ERROR(_append(_write_buf));
+            _write_buf.reset();
+        }
+    }
+    // need to allocate a new chunk, min chunk is 64k
+    size_t chunk_size = std::max(_min_chunk_size, size - pos);
+    chunk_size = BitUtil::RoundUpToPowerOfTwo(chunk_size);
+    _write_buf = ByteBuffer::allocate(chunk_size);
+    _write_buf->put_bytes(data + pos, size - pos);
+    return Status::OK();
+}
+
+StatusOr<ByteBufferPtr> StreamLoadPipe::read() {
+    if (_use_pipeline) {
```

Comment:
Could you just name it as `non_blocking_read` insteaf od `use_pipeline`?

Reply:
ok